File: examples/python/basic_request.py
"""
Salting API — Python example
Call any private API securely without a backend.

Docs: https://salting.io/docs
Get started: https://salting.io
"""

import requests

BRIDGE_URL = "https://api.salting.io/r/your-bridge-uuid"


def secure_get(params: dict = {}):
    """
    Secure GET request via Salting bridge.
    
    Args:
        params: query parameters, e.g. {"limit": "10"}
    """
    response = requests.get(BRIDGE_URL, params=params)
    response.raise_for_status()
    return response.json()


def secure_post(body: dict = {}):
    """
    Secure POST request via Salting bridge.

    Args:
        body: request body, e.g. {"model": "gpt-4", "messages": [...]}
    """
    response = requests.post(BRIDGE_URL, json=body)
    response.raise_for_status()
    return response.json()


if __name__ == "__main__":
    # GET — e.g. Stripe prices
    prices = secure_get(params={"limit": "10"})
    print("Prices:", prices)

    # POST — e.g. OpenAI chat
    chat = secure_post(body={
        "model": "gpt-4",
        "messages": [
            {"role": "user", "content": "Hello!"}
        ]
    })
    print("Chat:", chat)

File: examples/javascript/basic_request.js
/**
 * Salting API — Basic request example
 * Call any private API securely without a backend.
 *
 * Docs: https://salting.io/docs
 * Get started: https://salting.io
 */

const BRIDGE_URL = "https://api.salting.io/r/your-bridge-uuid";

// GET request
async function secureGet(params = {}) {
  const url = new URL(BRIDGE_URL);
  Object.entries(params).forEach(([k, v]) => url.searchParams.append(k, v));

  const response = await fetch(url.toString());

  if (!response.ok) {
    throw new Error(`Request failed: ${response.status}`);
  }

  return response.json();
}

// POST request
async function securePost(body = {}) {
  const response = await fetch(BRIDGE_URL, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(body),
  });

  if (!response.ok) {
    throw new Error(`Request failed: ${response.status}`);
  }

  return response.json();
}

// Examples
async function main() {
  // GET — e.g. Stripe prices
  const prices = await secureGet({ limit: "10" });
  console.log("Prices:", prices);

  // POST — e.g. OpenAI chat
  const chat = await securePost({
    model: "gpt-4",
    messages: [{ role: "user", content: "Hello!" }],
  });
  console.log("Chat:", chat);
}

main().catch(console.error);
